boot/freeldr/freeldr/arch/i386/pcmem.c

```diff
@@ -354,6 +354,12 @@ PcMemGetBiosMemoryMap(PFREELDR_MEMORY_DESCRIPTOR MemoryMap, ULONG MaxMemoryMapSi
         {
             ERR("PcMemoryMap is already full! (PcBiosMapCount = %lu, PcMapCount = %lu (>= %lu))\n",
                 PcBiosMapCount, PcMapCount, MaxMemoryMapSize);
+            ASSERTMSG("PcMemoryMap is already full!", FALSE);
```

Comment:
This can be good or bad. Yes, it would catch the case where we make a broken assumption and inform us about it. But it might also increase the number of systems on which ROS "doesn't work."
I agree with this method for new code, but for existing code, "don't break what works" weighs stronger IMO, unless you've actually encountered a board with this problem (indicating that there's something that will need fixing soon anyway).

Reply:
In an ideal world, we'd have a error reporting/notification system, where FreeLdr can place information in the registry and once booted, reactos would show it.

Reply:
@ThFabba,
I would usually agree, except, in this function, some of the missed entries could be overlapping with a different (more restrictive) type ... hence the (ASSERT +) all-or-nothing fixmes (which I plan to fix later, in the same issue) I'm adding fttb.
Current code is not simply missing some usable RAM (hopefully), it could lead to a later misbehavior (in edge cases).
I don't have an actual case, I hope our limits are high enough to never be reached, but I still suggest "better safe than sorry".
(Out-of-scope: CONTinuing after these ASSERT, using REL builds instead of DBG, ...)

@tkreuzer,
exactly what I wished about 👍

Reply:
I don't have a strong opinion, by am leaning towards "don't touch a working system." @tkreuzer @HBelusca?